src/modules/dashboard/components/Dialogs/ControlSafeDialog/SafeTransactionPreview.tsx

```diff
@@ -242,36 +310,39 @@ const SafeTransactionPreview = ({ colony, values }: Props) => {
                       />
                     </div>
                   )}
-                <DetailsItem
-                  label={MSG.safe}
-                  value={
-                    <AddressDetailsView
-                      item={(values.safe as never) as AnyUser}
-                      isSafeItem
-                    />
-                  }
-                />
-                {values.transactions[index].transactionType !==
-                  TransactionTypes.CONTRACT_INTERACTION && (
+                {!isNFT(index) && (
                   <DetailsItem
-                    label={MSG.to}
+                    label={MSG.safe}
                     value={
                       <AddressDetailsView
-                        item={
-                          (values.transactions[index]
-                            .recipient as never) as AnyUser
-                        }
-                        isSafeItem={false}
+                        item={(values.safe as never) as AnyUser}
+                        isSafeItem
                       />
                     }
                   />
                 )}
+                {values.transactions[index].transactionType !==
+                  TransactionTypes.CONTRACT_INTERACTION &&
+                  !isNFT(index) && (
+                    <DetailsItem
+                      label={MSG.to}
+                      value={
+                        <AddressDetailsView
+                          item={
+                            (values.transactions[index]
+                              .recipient as never) as AnyUser
+                          }
+                          isSafeItem={false}
+                        />
+                      }
+                    />
+                  )}
                 {values.transactions[index].transactionType !== '' &&
                   transactionTypeFieldsMap[
                     values.transactions[index].transactionType
                   ].map(({ key, label, value }) => (
                     <DetailsItem
-                      key={key}
+                      key={`${key}${label.defaultMessage}`}
```

Comment:
I believe we could remove the key prop from `transactionTypeFieldsMap` and make this key better if we use autogenerated ids. Check out how we `nanoid` to generate ids in cases like this where there's no clear way to get a unique id.